### PR TITLE
Change Log Updates

### PR DESCRIPTION
**Pull request by stuartarchibald**
Updates to the ChangeLog for 0.34, feel free to fork/comment/alter etc.

### CODE REVIEW

**Comment by seibert** [comment]
Can you mention that `parallel` is not available on Windows Python 2.7 for this release?

**Comment by stuartarchibald** [comment]
Good point, done in https://github.com/numba/numba/pull/2430/commits/6c66e35683bdd7ab7d99bf7a04e746a6507e02fc

File: CHANGE_LOG [hunk 1]
```diff
@@ -1,7 +1,90 @@
...
+thread-safety, type inference and other smaller enhancements and bugs have been
+fixed.
+
+
+ParallelAccelerator features:
+
+NOTE: The ParallelAccelerator technology is under active development and should
+      be considered experimental.
+
+The ParallelAccelerator technology is accessed via a new no Python mode option
```
Comment (seibert) [inline CHANGE_LOG]: I would write this a "nopython" mode, to match the other docs.

Comment (stuartarchibald) [inline CHANGE_LOG]: Good idea, done in  4e1458a.

**Comment by stuartarchibald** [comment]
Thanks for review @seibert. Will wait for a build pass then merge and tag for RC.

**Pull request merged**